openjpa-persistence/src/main/java/org/apache/openjpa/persistence/XMLPersistenceMetaDataParser.java

```diff
@@ -575,17 +577,25 @@ protected Object getSchemaSource() {
         boolean useExtendedSchema = true;
         // if the version and/or schema location is for 1.0, use the 1.0
         // schema
-        if (_ormVersion != null &&
-            _ormVersion.equals(XMLVersionParser.VERSION_1_0) ||
-            (_schemaLocation != null &&
-            _schemaLocation.indexOf(ORM_XSD_1_0) != -1)) {
+        if (XMLVersionParser.VERSION_1_0.equals(_ormVersion)
+                || (_schemaLocation != null && _schemaLocation.indexOf(ORM_XSD_1_0) > -1)) {
             ormxsd = "orm-xsd.rsrc";
             useExtendedSchema = false;
+        } else if (XMLVersionParser.VERSION_2_0.equals(_ormVersion)
+                || (_schemaLocation != null && _schemaLocation.indexOf(ORM_XSD_2_0) > -1)) {
+            ormxsd = "orm_2_0-xsd.rsrc";
+        } else if (XMLVersionParser.VERSION_2_1.equals(_ormVersion)
+                || (_schemaLocation != null && _schemaLocation.indexOf(ORM_XSD_2_1) > -1)) {
+            ormxsd = "orm_2_1.xsd.rsrc";
+            useExtendedSchema = false;
```

Comment:
I'm not sure about this

Reply:
I'd put it true to be able to use openjpa namespace, any reason to not do?
Also think it is worth some tests otherwise there is no real code change/feature ;)

Reply:
@rmannibucau IMO the main PRO is to be able to work completely offline
I have no idea how to write test like this :(

Reply:
use the JVM HttpServer, mock an xsd you use in the test persistence.xml and ensure the server handler is not hit during an emf->em creation (+close), will be sufficient.

Reply:
@rmannibucau  I hope these tests will be OK :)

Reply:
@rmannibucau is this one OK to be merged? :)